Add link in salsa (#1866)
rust-lang/rustc-dev-guide: src/salsa.md

> As of <!-- date-check --> November 2022, although Salsa is inspired by
> (among other things) rustc's query system, it is not used directly in rustc.

> > It _is_ used in [chalk], the library for the trait system, and extensively in

Reviewer: ```suggestion
> It _is_ used in [chalk], an implementation of  Rust's trait system, and extensively in
```

or something like that. “The library” sounds like it's the only library and actively used. However, Chalk is actually no longer used in the compiler (it used to be available under `-Zchalk` and later `-Ztrait-solver=chalk` before it got ripped out).

> > [`rust-analyzer`], the implementation of the language server protocol, but

Reviewer: ```suggestion
> [`rust-analyzer`], the official implementation of the language server protocol for Rust, but
```

There are other LSP-based language servers, e.g. https://github.com/intellij-rust/intellij-rust.
